#!/usr/bin/env python

PACKAGE = 'vpr_tracking'

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("averaging_buffer_size", int_t, 0, "Pose History buffer size, set to 0 to skip averaging", 0, 0, 200)
gen.add("weight_coeff", double_t, 0, "Averaging coefficient, larger values make the latest pose more influential", 0.5, 0.0, 1.0)
gen.add("min_num_markers", int_t, 0, "Minimum number of markers required to estimate a tool pose", 2, 1, 5)
gen.add("max_allowed_final_cost", double_t, 0, "Max allowed cost after optimizer convergence for a good pose estimate (-log10)",
    2.0, 0.0, 5.0)

exit(gen.generate(PACKAGE, PACKAGE, "ToolTracker"))
